gms/gossiper.cc

```diff
@@ -1099,18 +1099,18 @@ future<> gossiper::advertise_token_removed(inet_address endpoint, utils::UUID ho
         endpoint_state_map[endpoint] = eps;
         replicate(endpoint, eps).get();
         // ensure at least one gossip round occurs before returning
-        sleep(INTERVAL * 2).get();
+        sleep_abortable(INTERVAL * 2, as).get();
     });
 }
 
-future<> gossiper::unsafe_assassinate_endpoint(sstring address) {
+future<> gossiper::unsafe_assassinate_endpoint(sstring address, abort_source& as) {
     logger.warn("Gossiper.unsafeAssassinateEndpoint is deprecated and will be removed in the next release; use assassinate_endpoint instead");
-    return assassinate_endpoint(address);
+    return assassinate_endpoint(address, as);
 }
 
-future<> gossiper::assassinate_endpoint(sstring address) {
-    return get_gossiper().invoke_on(0, [address] (auto&& gossiper) {
-        return seastar::async([&gossiper, g = gossiper.shared_from_this(), address] {
+future<> gossiper::assassinate_endpoint(sstring address, abort_source& as) {
+    return get_gossiper().invoke_on(0, [address, &as] (auto&& gossiper) {
+        return seastar::async([&gossiper, g = gossiper.shared_from_this(), address, &as] {
```

Comment:
`abort_source` is not safe to use across shards (across threads). It's a shard-local object.

invoke_on() may execute the lambda on a different shard if the current shard is not 0.

Reply:
I removed the `abort_source&` argument from the function. It wouldn't be used anyway (as we don't handle cancellation of HTTP requests), and I can't think of a solution for passing the abort source that would look nice.